spark/src/main/scala/com/gu/tableversions/spark/VersionContext.scala

```diff
@@ -87,6 +73,27 @@ object VersionedDataset {
     } yield (latestTableVersion, metastoreUpdate)
   }
 
+  object implicits {
+
+    implicit class DatasetOps[T](val delegate: Dataset[T]) {
+
+      /**
+        * Insert the dataset into the given versioned table.
+        *
+        * This emulates the behaviour of Hive inserts in that it will overwrite any partitions present in the dataset,
+        * while leaving other partitions unchanged.
+        *
+        * @return a tuple containing the updated table version information, and a list of the changes that were applied
+        *         to the metastore.
+        */
+      def versionedInsertInto(table: TableDefinition, userId: UserId, message: String): (TableVersion, TableChanges) =
+        versionedInsertDatasetIntoTable(delegate, table, userId, message).unsafeRunSync()
+    }
+  }
+}
+
+object VersionContext {
```

Comment:
As a slight aside, I think there are some more `implicit`s you can get rid of in this code if you want, on the `partitionValues` and `writeVersionedPartitions` method?